src/app/components/shared/breadcrumb/breadcrumb.component.ts

```diff
@@ -0,0 +1,77 @@
+import { Component, OnInit } from "@angular/core";
+import { Params } from "@angular/router";
+import { withUnsubscribe } from "@helpers/unsubscribe/unsubscribe";
+import { BreadcrumbsData, MenuService } from "@services/menu/menu.service";
+import { SharedActivatedRouteService } from "@services/shared-activated-route/shared-activated-route.service";
+import { Observable, takeUntil } from "rxjs";
+
+/**
+ * Menu Link Component
+ */
+@Component({
+  selector: "baw-breadcrumbs",
+  template: `
+    <nav *ngIf="shouldShowBreadcrumbs()" aria-label="breadcrumb">
+      <ol class="breadcrumb bg-light p-1">
+        <li
+          *ngFor="let breadcrumb of breadcrumbs; let i = index"
+          class="breadcrumb-item"
+          [class.active]="isLastBreadcrumb(i)"
+        >
+          <fa-icon
+            class="pe-1"
+            [class.text-highlight]="!isLastBreadcrumb(i)"
+            [icon]="breadcrumb.icon"
+          ></fa-icon>
+          <a
+            *ngIf="!isLastBreadcrumb(i)"
+            [strongRoute]="breadcrumb.route"
+            [routeParams]="routeParams | async"
+            [queryParams]="queryParams | async"
+          >
+            {{ breadcrumb.label }}
+          </a>
+          <span *ngIf="isLastBreadcrumb(i)">{{ breadcrumb.label }}</span>
+        </li>
+      </ol>
+    </nav>
+  `,
+})
+export class BreadcrumbComponent extends withUnsubscribe() implements OnInit {
+  public queryParams: Observable<Params>;
+  public routeParams: Observable<Params>;
+  public hasRouteData: boolean;
+  public breadcrumbs: BreadcrumbsData;
+
+  public constructor(
+    private sharedRoute: SharedActivatedRouteService,
+    private menu: MenuService
+  ) {
+    super();
+  }
+
+  public ngOnInit(): void {
+    this.menu.menuUpdate
+      .pipe(takeUntil(this.unsubscribe))
+      .subscribe(({ breadcrumbs }): void => {
+        this.breadcrumbs = breadcrumbs;
+      });
```

Comment:
this is a good candidate for async pipes - the logic here only deals with unsubscribes  and ~~the state is not used elsewhere in the class~~ (ok it is, but that's what a query function is for right?)